tdrs-backend/tdpservice/parsers/util.py

```diff
@@ -236,15 +236,20 @@ def add_record(self, case_hash, record_doc_pair, line_num):
             logger.error(f"Error: Case hash for record at line #{line_num} was None!")
 
     def get_bulk_create_struct(self):
-        """Return dict of form {document: Iterable(records)} for bulk_create_records to consume."""
+        """Return dict of form {document: {record: None}} for bulk_create_records to consume."""
         return self.cases
 
     def clear(self, all_created):
         """Reset sorted structs if all records were created."""
         if all_created:
             self.serialized_cases.update(set(self.hash_sorted_cases.keys()))
             self.hash_sorted_cases = dict()
-            self.cases = dict()
+
+            # We don't want to re-assign self.cases here because we lose the keys of the record/doc types we've already
+            # made. If we don't maintain that state we might not delete everything if we need to roll the records back
+            # at the end of, or during parsing.
+            for key in self.cases.keys():
+                self.cases[key] = {}
```

Comment:
Great comment explaining why we need to delete just the contents but not the keys/structure.